Classes/EventSourcedRepository.php

```diff
@@ -76,7 +73,7 @@ public function save(AggregateRootInterface $aggregate)
         } catch (EventStreamNotFoundException $e) {
             $stream = new EventStream(
                 $aggregate->getAggregateIdentifier(),
-                get_class($aggregate),
+                TypeHandling::getTypeForValue($aggregate),
```

Comment:
Didn't mention this in the commit message, but we should avoid `get_class` for any object that can be proxied!